### ISSUE TITLE
Click to enlarge link

### ISSUE THREAD

**Issue by UseDotCom** [UseDotCom/lighthouse#752]
Created by **[usedotcom](https://github.com/usedotcom)**  at 18:17Z on 06/16/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

In the HTML images code provided in the gencodes box, the link doubles up the http... and, so, throws a 404

Example for set eid 764949f710de7bcbec1d (titled Referrer - html layout - Monday - 06):

@@@ html
<center><a href="http://www.desktopcurrency.com/Bex" target="_blank"><img border="0" src="http://www.desktopcurrency.com/images/s_2/764949f710de7bcbec1d_1.jpg" height="315"></a><br>
<a href="http://www.desktopcurrency.com/http://www.desktopcurrency.com/Bex?p=1" target="_blank"><img src="http://www.desktopcurrency.com/images/clicklarge3.gif" border="0"></a><br>
<br>
<a href="http://www.desktopcurrency.com/Bex" target="_blank"><img border="0" src="http://www.desktopcurrency.com/images/s_2/764949f710de7bcbec1d_2.jpg" width="420"></a><br>
<a href="http://www.desktopcurrency.com/http://www.desktopcurrency.com/Bex?p=2" target="_blank"><img src="http://www.desktopcurrency.com/images/clicklarge3.gif" border="0"></a></center>
@@@

- - - -

Lighthouse ticket URL: http://use-com.lighthouseapp.com/projects/103799/tickets/334

Attachments:

* https://use-com.lighthouseapp.com/attachments/5150422/embedcodes-typo.png
* https://use-com.lighthouseapp.com/attachments/5150437/error-message-supersize.png
* https://use-com.lighthouseapp.com/attachments/5150438/error-message-viewer.png

Was assigned to **[usedotcom](https://github.com/usedotcom)**

**Comment by UseDotCom** [comment]
By **Dave** at 15:18Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

I forgot to reply to this last night (it was too late to call).

Which instance of gencodes is broken? There are multiple different places in the site that either externally or internally refer to themselves as 'gencodes'.

**Comment by UseDotCom** [comment]
By **Dave** at 15:57Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

I've found another issue where editset is the referenced URL instead of the standard url in this box. I'm tagging it in this ticket as it requires changes to the same code.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 17:14Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Are you sure it's actually referencing editset in the generated code and that you're not just being redirected because you are the owner?

**Comment by UseDotCom** [comment]
By **Dave** at 17:30Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Positive. It's in the database with editset.

**Comment by UseDotCom** [comment]
By **Dave** at 22:45Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Fixed.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 23:14Z on 06/17/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Currently, on Use.com, when you use the embed codes, if you click on an image in a set other than the first one, the viewset page will display with that image's s_2 in the showcase display. This is particularly noticeable when working with the HTML images code, because the Click to enlarge link displays directly below an image, but that is not the showcased image when you click.

Is it possible to retain this feature while using the tiny url's? If not, is it feasible to use the full eid for actual embed codes and tiny url's for direct links (ie the photo set URL)? What are the downsides to doing this?

**Comment by UseDotCom** [comment]
By **Dave** at 03:57Z on 06/18/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

This is feasible. The downside comes from database access as each individual photo has to have its own short URL. I should be able to put this together in about a day.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 13:21Z on 06/18/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

To clarify and help us make a rational choice:

1. Is it easier and/or less cumbersome to use the full eid on the HTML embed codes and the tiny url's on the links where the user would actually be giving out the URL instead of pasting HTML into a listing? Are there downsides to doing *this*?

2. If we create tiny url's for each individual image, will that negatively affect speed or push us to significantly longer url's much sooner?

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 15:05Z on 06/18/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Spoke with Jonathan. Preferred solution is using full eid in embed codes, tiny urls in direct link urls. This will preserve the available tiny urls and there really isn't any advantage to having them in the HTML code anyway.

**Comment by UseDotCom** [comment]
By **Dave** at 02:19Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Ok, made the change. All embed codes are full urls.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 13:51Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Again, lack of live testing.

In the gencodes pop up code box (Profile>Generate HTML &Image codes), the embed codes have relative URL's rather than absolute and use the editset.pl, which is incorrect. If these codes are embedded, they, as expected, throw a 404 error.

Another place where the code needs to be reverted to use full eids rather than tiny urls is in the viewer page (example http://www.desktopcurrency.com/9577db0e1c6559fa2f2d#photo=2). Click the Show links & HTML codes below the showcase. The URL correctly displays the tiny url for the set. The HTML and IMG embed codes should provide a link to the set with the selected photo as the showcase. The embed codes currently work this way on Use.com, which can be used as a reference.

The neweid pop up code box (upload photos, you will receive the code box)still shows tiny urls in all codes. I also just noticed that only the first image is included in the code for thumbnails and html images. Is this an unavoidable issue due to providing the codes immediately? Can we grab the count from the upload form so all images are included? Or is there another fix? I'd rather have a broken link in the rare instances that an upload fails than to be generating the wrong code.

**Comment by UseDotCom** [comment]
By **Dave** at 20:02Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Gencodes is working. Working on the viewer page.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 20:18Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Gencodes now has a typo in the URL. Missing the period after www.

The viewer page has been reverted to the full eid but has lost the functionality referenced. If you have selected the 4th photo and grab the code, it should link to the viewer page with the 4th photo in the showcase position. It currently does this on Use.com. Please look at it, if this is not clear. Here is an example: http://www.use.com/Corelle_Livingware_16_piece_Dinnerware_Set_Flor_d9ecb2e9ad8ba9cb90bb#photo=1

**Comment by UseDotCom** [comment]
By **Dave** at 20:19Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

I haven't done anything to the viewer page.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 20:27Z on 06/19/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

You're right. I read your update and thought it was done. I missed that the tiny url was still being used and thought the only remaining issue with it was the photo selection.

**Comment by UseDotCom** [comment]
By **Dave** at 02:44Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Typo fixed and viewset is also done. I also fixed an issue where swapping images would cause an error dialog to show up.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 14:16Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Typo still displaying. See attached.

I was apparently mistaken about Use.com returning the correct link on the Viewer page. It is also not indicating the correct showcase. I am moving this to a separate ticket.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 14:17Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Am also now receiving the attached error message whenever I visit a SuperSize link or when I click through images on the Viewer page. The Viewer page gives the error whether using the arrow or clicking on a thumb. This did not occur previously, so presumably is related either to recent changes or hardware issues. See attached

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 15:14Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

This may be what is throwing the error message. When you click through the images on the viewer page, the URL text box changes to "Undefined"
I'm guessing this is because of the tiny url's. On Use.com, it does give the url of the set with the specified image as the Showcase.  I'm waiting on an answer from Jonathan as to how he wants to handle this.

James made a step by step of what he was experiencing:

http://james.macksracks.net/use/undefinedlink/

**Comment by UseDotCom** [comment]
By **Dave** at 15:22Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Sorry about that. There were quite a number of files which needed deploying last night, and I missed the two related to this change. Try now.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 18:27Z on 06/20/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

Typos are fixed.

As for the URL code box on the Viewer page, in the "Show links & IMG code" drop down div, the error messages still display as described.
It appears that there was an attempt to designate the selected image as the Showcase, but it is failing somewhere.

It seems that, even  though we decided to not use the tiny urls in embed codes, new tiny urls are being generated for each embed code link, as they are visited. 

I believe the easiest fix for now is to simply use the set's tiny URL for all images in the URL code box on the Viewer page Show links & IMG code drop down div. This can use the same tiny URL that is displayed as the Direct URL in the Gencodes box. Please advise.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 20:57Z on 06/23/2014 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/334))*

- - - -

As discussed, generating a tiny url for each photo on demand is acceptable as we would definitely be able to keep max characters to no more than 6.

**Comment by UseDotCom** [comment]
Imported from Lighthouse  at 21:10Z on 02/05/2015

Assigned to ticket creator, **[usedotcom](https://github.com/usedotcom)**
